lib/rucio/core/rule.py

```diff
@@ -2040,15 +2412,41 @@ def update_rules_for_bad_replica(scope, name, rse_id, nowait=False, *, session:
         else:
             rule.state = RuleState.REPLICATING
             if rule.grouping != RuleGrouping.NONE:
-                session.query(models.DatasetLock).filter_by(rule_id=rule.id).update({'state': LockState.REPLICATING})
+                stmt = update(
+                    models.DatasetLock
+                ).where(
+                    models.DatasetLock.rule_id == rule.id
+                ).values(
+                    {'state': LockState.REPLICATING}
+                )
+                session.execute(stmt)
         # Insert rule history
         insert_rule_history(rule=rule, recent=True, longterm=False, session=session)
     if nlock:
-        session.query(models.RSEFileAssociation).filter(models.RSEFileAssociation.scope == scope, models.RSEFileAssociation.name == name, models.RSEFileAssociation.rse_id == rse_id).update({'state': ReplicaState.COPYING})
+        stmt = update(
+            models.RSEFileAssociation
+        ).where(
+            and_(models.DataIdentifier.scope == scope,
+                 models.RSEFileAssociation.rse_id == rse_id,
+                 models.DataIdentifier.name == name)
+        ).values(
+            {'availability': ReplicaState.COPYING}
```

Comment:
I think this should be `state`, not `availability`?

Reply:
You're right. Changed now to use state instead